docs/quat2.js.html

```diff
@@ -0,0 +1,948 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+    <title>JSDoc: Source: quat2.js</title>
+
+    <script src="scripts/prettify/prettify.js"> </script>
+    <script src="scripts/prettify/lang-css.js"> </script>
+    <!--[if lt IE 9]>
+      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
+    <![endif]-->
+    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
+    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
+</head>
+
+<body>
+
+<div id="main">
+
+    <h1 class="page-title">Source: quat2.js</h1>
+
+    
+
+
+
+    
+    <section>
+        <article>
+            <pre class="prettyprint source linenums"><code>/* Copyright (c) 2015, Brandon Jones, Colin MacKenzie IV.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE. */
+
+//TODO: TOTALLY FIX ME!! (Newer function signatures, ES5 modules?)
+//Remove every var --> let
+
+import * as glMatrix from "./common";
+import * as quat from "./quat";
+import * as mat4 from "./mat4";
+
+//NOTE: Quaternions are in XYZW format!
+//NOTE: [real dual]
+//NOTE: Dual Quaternions need to be normalized, otherwise some functions won't work!
+//TODO: slerp and co.
+//TODO: rotate(out, a, rad, axis) 
+//TODO: quat2.forEach
+
+//LINKS: https://github.com/bobbens/libdq
+//https://github.com/markaren/DualQuaternion/tree/master/src/main/java/info/laht/dualquat
+//http://wscg.zcu.cz/wscg2012/short/A29-full.pdf
+//http://www.xbdev.net/misc_demos/demos/dual_quaternions_beyond/paper.pdf
+//https://github.com/mosra/magnum/blob/master/src/Magnum/Math/DualQuaternion.h#L364
+//http://mosra.cz/blog/magnum-doc/classMagnum_1_1Math_1_1DualQuaternion.html#a6dcad20c0e9c7c9361b9c291ee42b181
+/**
+ * @class Dual Quaternion&lt;br>Format: [[real],[dual]]&lt;br>Make sure to have normalized dual quaternions, otherwise the functions may not work as intended.
```

Comment:
confused here isn't the format `[real, dual]` ?

Reply:
Yeah, you are right